src/Carnac.Logic/KeyProvider.cs

```diff
@@ -101,10 +106,19 @@ KeyPress ToCarnacKeyPress(InterceptKeyEventArgs interceptKeyEventArgs)
                            interceptKeyEventArgs.Key <= Keys.Z;
 
             var inputs = ToInputs(isLetter, winKeyPressed, interceptKeyEventArgs);
+            try
+            {
+                Icon icon = Icon.ExtractAssociatedIcon(process.MainModule.FileName);
```

Comment:
It would be good if we cached the result of this, i'm not sure how heavy the call is but this could allocate a lot of duplicated Icons in memory?